ray-operator/config/samples/ray-job.cluster-selector.yaml

```diff
@@ -0,0 +1,149 @@
+              - mountPath: /home/ray/samples
+                name: code-sample
+        volumes:
+          - name: code-sample
+            configMap:
+              name: ray-job-code-sample
+              items:
+                - key: sample_code.py
+                  path: sample_code.py
+                - key: fail_fast.py
+                  path: fail_fast.py
+  workerGroupSpecs:
+    # the pod replicas in this group typed worker
+    - replicas: 1
+      minReplicas: 1
+      maxReplicas: 5
+      # logical group name, for this called small-group, also can be functional
+      groupName: small-group
+      # The `rayStartParams` are used to configure the `ray start` command.
+      # See https://github.com/ray-project/kuberay/blob/master/docs/guidance/rayStartParams.md for the default settings of `rayStartParams` in KubeRay.
+      # See https://docs.ray.io/en/latest/cluster/cli.html#ray-start for all available options in `rayStartParams`.
+      rayStartParams: {}
+      #pod template
+      template:
+        spec:
+          containers:
+            - name: ray-worker # must consist of lower case alphanumeric characters or '-', and must start and end with an alphanumeric character (e.g. 'my-name',  or '123-abc'
+              image: rayproject/ray:2.7.0
+              lifecycle:
+                preStop:
+                  exec:
+                    command: [ "/bin/sh","-c","ray stop" ]
+              resources:
+                limits:
+                  cpu: "1"
+                requests:
+                  cpu: "1"
+---
+# This sample is from https://docs.ray.io/en/latest/cluster/job-submission.html#quick-start-example.
+# It is mounted into the container and executed to show the Ray job at work
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: ray-job-code-sample
+data:
+  sample_code.py: |
+    import ray
+    import os
+    import requests
+
+    ray.init()
+
+    @ray.remote
+    class Counter:
+        def __init__(self):
+            # Used to verify runtimeEnv
+            self.name = os.getenv("counter_name")
+            assert self.name == "test_counter"
+            self.counter = 0
+
+        def inc(self):
+            self.counter += 1
+
+        def get_counter(self):
+            return "{} got {}".format(self.name, self.counter)
+
+    counter = Counter.remote()
+
+    for _ in range(5):
+        ray.get(counter.inc.remote())
+        print(ray.get(counter.get_counter.remote()))
+
+    # Verify that the correct runtime env was used for the job.
+    assert requests.__version__ == "2.26.0"
+  fail_fast.py: |
+    import sys
+
+    print >> sys.stderr, "Something is seriously wrong."
+    sys.exit(1)
+---
+# The RayJob, that references the RayCluster into which the Ray job is to be submitted.
+# You can change the entrypoint to fail_fast.py for testing the case where the job fails.
+apiVersion: ray.io/v1
+kind: RayJob
+metadata:
+  name: rayjob-cluster-selector
+spec:
+  clusterSelector:
```

Comment:
We do not encourage users to use `clusterSelector`. The better option is to use `ray job submit` to submit a ray job to a RayCluster.